portfolio/src/main/java/com/google/sps/servlets/models/CommentReader.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.servlets.models;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.datastore.TransactionOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public abstract class CommentReader {
+  private final static Logger LOGGER = Logger.getLogger(CommentReader.class.getName());
```

Comment:
private static final, not private final static.

Also again this is something you don't want to be a singleton.  Just declare it as a class variable.  In this case, logger is single-threaded, so if you had 250,000 people coming into your website and all trying to log with a single instance of logger it would become the bottleneck.